### DISCUSSION TITLE
pnpm + vitest + esm+ MikroORM.init = Unknown file extension ".ts"

### DISCUSSION THREAD

**Discussion by secato** [mikro-orm/mikro-orm#5517] [General]
### Describe the bug

I'm in a ESM project and trying to call `MikroORM.init()` in a test suite but I'm getting:
```
FAIL  src/main.spec.ts > MikroORM init
TypeError: Unknown file extension ".ts" for /home/secato/dev/tsissue/src/mikro-orm.config.ts
```

Debug output

```
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 6.2.3
   - node 22.0.0
   - knex 3.1.0
   - sqlite3 5.1.7
   - typescript 5.4.5
 - package.json found
 - ts-node enabled
 - searched config paths:
   - /home/secato/dev/tsissue/src/mikro-orm.config.ts (found)
   - /home/secato/dev/tsissue/mikro-orm.config.ts (not found)
   - /home/secato/dev/tsissue/src/mikro-orm.config.js (not found)
   - /home/secato/dev/tsissue/mikro-orm.config.js (not found)
 - configuration found
 - database connection successful
 - will use `entities` array (contains 0 references and 1 paths)
   - /home/secato/dev/tsissue/dist/**/*.entity.js (not found)
 - could use `entitiesTs` array (contains 0 references and 1 paths)
   - /home/secato/dev/tsissue/src/**/*.entity.ts (found)
```

**To reproduce:**
- clone the repo below
- pnpm i
- pnpm test

### Reproduction

https://github.com/secato/mikro-esm-issue

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

6.2.3

### Node.js version

22

### Operating system

Ubuntu 22.04 via wsl2

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
https://mikro-orm.io/docs/guide/project-setup#testing-the-endpoint

**Reply by secato** [reply]
No way I completely missed this, sorry man I completely missed this. 
First time trying the ESM/Typescript setup thing in a side project.

You did great with the docs, even though this topic of typescript + esm + transpilers + test is supeeeeer messy.

Closing this, thanks again for the prompt reply and for all you work here, this is the best ORM for node!!!!!
